core/templates/dev/head/pages/exploration_editor/simple_editor_tab/questions/TextInputCheckerServiceSpecs.js

```diff
@@ -0,0 +1,85 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for TextInteractionInputChecker.
+ */
+ describe('Text interaction input checker service', function() {
+   beforeEach(module('oppia'));
+
+   beforeEach(inject(function($injector) {
+     AnswerGroupObjectFactory = $injector.get('AnswerGroupObjectFactory');
+     tics = $injector.get('TextInputCheckerService');
+   }));
+   
+   var AnswerGroupObjectFactory;
+   var tics;
+
+   it('should return true for rule type EqualsS ', function() {
+     var answerGroupsTrue = [{
+       rules: [{
+         type: 'Equals',
+         inputs: {
+           x: 'j'
+         }
+       }],
+       outcome: {
+         param_changes: [],
+         feedback: [],
+         dest: 'Question 1'
+       }
+     }];
+     expect(tics.isValid('', answerGroupsTrue)).toBe(true);
```

Comment:
Hmm, I think you should put in some realistic `customizationArg` argument so that the types are being consistent.